src/mame/msx/msx.cpp

```diff
@@ -432,53 +428,16 @@ void msx_state::driver_start()
 	save_item(NAME(m_keylatch));
 }
 
-INTERRUPT_GEN_MEMBER(msx_state::msx_interrupt)
-{
-	m_mouse[0] = m_io_mouse[0]->read();
-	m_mouse_stat[0] = -1;
-	m_mouse[1] = m_io_mouse[1]->read();
-	m_mouse_stat[1] = -1;
-}
-
-template <u8 Game_port>
-u8 msx_state::game_port_r()
-{
-	u8 inp = m_io_joy[Game_port]->read();
-	if (!(inp & 0x80))
-	{
-		// joystick
-		return (inp & 0x7f);
-	}
-	else
-	{
-		// mouse
-		u8 data = (inp & 0x70);
-		if (m_mouse_stat[Game_port] < 0)
-			data |= 0xf;
-		else
-			data |= ~(m_mouse[Game_port] >> (4 * m_mouse_stat[Game_port])) & 0x0f;
-		return data;
-	}
-}
-
 u8 msx_state::psg_port_a_r()
 {
 	u8 data = 0x80;
 	if (m_cassette)
-	{
 		data = (m_cassette->input() > 0.0038 ? 0x80 : 0);
-	}
 
-	if ((m_psg_b ^ m_io_dsw->read()) & 0x40)
-	{
-		// game port 2
-		data |= game_port_r<1>();
-	}
+	if (BIT(m_psg_b, 6))
+		data |= bitswap<8>(m_gen_port2->read(), 4, 7, 6, 5, 3, 2, 1, 0) & 0x3f;
```

Comment:
This can be simplified to `bitswap<6>(m_gen_port2->read(), 6, 5, 3, 2, 1, 0);`.

Reply:
Thanks, fixed both of them.